whois-query/src/main/java/net/ripe/db/whois/query/acl/AccessControlListManager.java

```diff
@@ -65,13 +72,18 @@ public AccessControlListManager(final DateTimeProvider dateTimeProvider,
         this.ssoTokenTranslator = ssoTokenTranslator;
         this.ipRanges = ipRanges;
         this.isSSOAccountingEnabled = isSSOAccountingEnabled;
+        this.jdbcRpslObjectSlaveDao = jdbcRpslObjectSlaveDao;
     }
 
-    public boolean requiresAcl(final RpslObject rpslObject, final Source source) {
+    public boolean requiresAcl(final RpslObject rpslObject, final Source source, final AccountingIdentifier accountingIdentifier) {
         if (source.isGrs()) {
             return false;
         }
 
+        if (!StringUtil.isNullOrEmpty(accountingIdentifier.getSsoToken()) && isUserOwnedObject(rpslObject, accountingIdentifier.getSsoToken())){
```

Comment:
Can you use username ? Why you need crowdToken? If you start using crowdToken , it will not be universal as oauth2 and apikey will not have sso token

Reply:
please read my comment below

``` 
I decided to add the ssoToken coming from the request header in here, as an extra user identification step

There is another option to just keep the userName (as it is) and call the auth service to get the related ssoToken (it is 1 to 1 relation). However, this would allow the user to bypass the counting, putting a wrong email-uuid relation...(we would be ignoring the UUID from the request).

This is needed because in the Auth Index table (which I need to use because of task requirement) we are storing "SSO " as value for Auth: attribute. So for fetching the maintainters belonging to that user we need the UUID and not the email.
```

For reverse query I need the UUID

Reply:
SSOToken is different then uuid , SSOToken gets translated into uuid and username via authservice client, I think you meant to uuid only?